processor/servicegraphprocessor/processor_test.go

```diff
@@ -0,0 +1,279 @@
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	semconv "go.opentelemetry.io/collector/semconv/v1.6.1"
+	"go.uber.org/zap/zaptest"
+)
+
+func TestProcessorStart(t *testing.T) {
+	// Create otlp exporters.
+	otlpConfig, mexp, texp := newOTLPExporters(t)
+
+	for _, tc := range []struct {
+		name            string
+		exporter        component.Exporter
+		metricsExporter string
+		wantErrorMsg    string
+	}{
+		{"export to active otlp metrics exporter", mexp, "otlp", ""},
+		{"unable to find configured exporter in active exporter list", mexp, "prometheus", "failed to find metrics exporter: 'prometheus'; please configure metrics_exporter from one of: [otlp]"},
+		{"export to active otlp traces exporter should error", texp, "otlp", "the exporter \"otlp\" isn't a metrics exporter"},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			// Prepare
+			exporters := map[config.DataType]map[config.ComponentID]component.Exporter{
+				config.MetricsDataType: {
+					otlpConfig.ID(): tc.exporter,
+				},
+			}
+			mHost := &mockHost{
+				GetExportersFunc: func() map[config.DataType]map[config.ComponentID]component.Exporter {
+					return exporters
+				},
+			}
+
+			// Create servicegraph processor
+			factory := NewFactory()
+			cfg := factory.CreateDefaultConfig().(*Config)
+			cfg.MetricsExporter = tc.metricsExporter
+
+			procCreationParams := componenttest.NewNopProcessorCreateSettings()
+			traceProcessor, err := factory.CreateTracesProcessor(context.Background(), procCreationParams, cfg, consumertest.NewNop())
+			require.NoError(t, err)
+
+			// Test
+			smp := traceProcessor.(*processor)
+			err = smp.Start(context.Background(), mHost)
+
+			// Verify
+			if tc.wantErrorMsg != "" {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestProcessorShutdown(t *testing.T) {
+	// Prepare
+	factory := NewFactory()
+	cfg := factory.CreateDefaultConfig().(*Config)
+
+	// Test
+	next := new(consumertest.TracesSink)
+	p := newProcessor(zaptest.NewLogger(t), cfg, next)
+	err := p.Shutdown(context.Background())
+
+	// Verify
+	assert.NoError(t, err)
+}
+
+func TestProcessorConsume(t *testing.T) {
+	cfg := &Config{
+		MetricsExporter: "mock",
+		Dimensions:      []string{"some-attribute", "non-existing-attribute"},
+	}
+
+	mockMetricsExporter := newMockMetricsExporter(func(md pmetric.Metrics) error {
+		return verifyMetrics(t, md)
+	})
+
+	processor := newProcessor(zaptest.NewLogger(t), cfg, consumertest.NewNop())
+
+	mHost := &mockHost{
+		GetExportersFunc: func() map[config.DataType]map[config.ComponentID]component.Exporter {
+			return map[config.DataType]map[config.ComponentID]component.Exporter{
+				config.MetricsDataType: {
+					config.NewComponentID("mock"): mockMetricsExporter,
+				},
+			}
+		},
+	}
+
+	assert.NoError(t, processor.Start(context.Background(), mHost))
+
+	td := sampleTraces()
+	assert.NoError(t, processor.ConsumeTraces(context.Background(), td))
```

Comment:
Semantically speaking, shouldn't this test shut down the processor? It might also be good to leave a comment stating that the assertions are part of the verifyMetrics function, called in the "prepare" phase of the unit test.

Reply:
Agree, with the suggestion. Updated.